pkg/controller/attach_detach.go

```diff
@@ -191,11 +192,22 @@ func (r *ReconcileAttachDetach) triggerAttach(ctx context.Context, azVolumeAttac
 				response, attachErr = r.attachVolume(cloudCtx, azVolumeAttachment.Spec.VolumeID, azVolumeAttachment.Spec.NodeName, azVolumeAttachment.Spec.VolumeContext)
 			}
 		}
+		pods, err := r.controllerSharedState.getPodsFromVolume(ctx, r.client, azVolumeAttachment.Spec.VolumeName)
+		if err != nil {
+			klog.Infof("failed to get pods for volume (%s). Error: %v", azVolumeAttachment.Spec.VolumeName, err)
+		}
+
 		// update AzVolumeAttachment CRI with the result of the attach operation
 		var updateFunc func(interface{}) error
 		if attachErr != nil {
 			klog.Errorf("failed to attach volume %s to node %s: %v", azVolumeAttachment.Spec.VolumeName, azVolumeAttachment.Spec.NodeName, attachErr)
 
+			if len(pods) > 0 {
+				for _, pod := range pods {
+					r.controllerSharedState.eventRecorder.Eventf(pod.DeepCopyObject(), v1.EventTypeWarning, consts.ReplicaAttachmentFailedEvent, "%s mount for volume %s could not attached to node %s", azVolumeAttachment.Spec.RequestedRole, azVolumeAttachment.Spec.VolumeName, azVolumeAttachment.Spec.NodeName)
```

Comment:
Do we want to raise `ReplicaAttachmentFailedEvent` only for the replica attachments or for all attachments? If latter, Should we change `ReplicaAttachmentFailedEvent` to `AttachmentFailedEvent`?

Reply:
Yeah, I think it makes sense to raise these events only for replica attachments as there are other K8s events indicating success/failure for primary attachments. Will add a condition for that.